### PR TITLE
New way to install flow

### PR DESCRIPTION
**Pull request by lucfisc**
Add a Dockerfile that can run with a desktop on your browser with every flow packages installed and ready to use

### CODE REVIEW

**Comment by eugenevinitsky** [comment]
+1, we should clarify local installation vs. non-local installation and ensure that developers install the local install

**Comment by AboudyKreidieh** [comment]
Actually, @lucfisc can u replace the table of contents with this?
```
.. contents:: Table of contents
```

**Comment by lucfisc** [comment]
@AboudyKreidieh  Done

File: docs/source/flow_setup.rst [hunk 1]
```diff
@@ -1,4 +1,22 @@
-Setup Instructions
+.. contents:: Table of contents
+
+1 - Local Installation
```
Comment (eugenevinitsky) [inline docs/source/flow_setup.rst]: The numbering scheme here is different than the numbering scheme below. Is this correct @lucfisc

File: docs/source/flow_setup.rst [hunk 2]
```diff
@@ -228,10 +246,48 @@ jobs from there.
...
+::
+    1° docker pull lucasfischerberkeley/flow_desktop
+    2° docker run -d -p 5901:5901 -p 6901:6901 -p 8888:8888 lucasfischerberkeley/flow_desktop
+    
+Go into your browser ( Firefox, Chrome, Safari)
+::
+    1° Go to http://localhost:6901/?password=vncpassword
+    2° Go to Applications and open Terminal Emulator
+    3° For sumo: Write python flow/examples/sumo/sugiyama.py and run it
+    4° For rllib : Write python flow/examples/rllib/stabilizing_the_ring.py and run it
```
Comment (eugenevinitsky) [inline docs/source/flow_setup.rst]: so rllib is not installed in a virtual env but rllab is?

**Comment by eugenevinitsky** [comment]
A few more minor comments, almost there!

File: docs/source/flow_setup.rst [hunk 3]
```diff
@@ -1,4 +1,22 @@
...
+    D. Testing the Installation
+    E. Rllab-multiagent (optional)
+    F. Ray/RLlib (optional)
+    G. Getting started (rllab-multiagent)
+    H. Getting started (Ray/RLlib)
+    I. Custom configuration
+
+2- Remote desktop using Docker (recommended for testing purpose)
+    a. Installation
+    b. Notebooks and tutorial
```
Comment (AboudyKreidieh) [inline docs/source/flow_setup.rst]: `.. contents:` will actually automatically generate the table of contents, so no need for this anymore

File: docs/source/flow_setup.rst [hunk 4]
```diff
@@ -228,10 +246,48 @@ jobs from there.
...
+9. Custom configuration
 ====================
 
 You may define user-specific config parameters as follows
 ::
 
     cp flow/core/config.template.py flow/core/config.py  # Create template for users using pycharm
+
+
+Remote desktop using Docker for simple utilisation of flow (recommended for testing purpose)
```
Comment (AboudyKreidieh) [inline docs/source/flow_setup.rst]: Add a "2." here

File: docs/source/flow_setup.rst (hunk 2 above)
Comment (lucfisc) [reply docs/source/flow_setup.rst]: @eugenevinitsky. Rllib is installed in the root environment and works.
For rllab, I need to enter another environment ( flow-rllab) but I am working on a way to have everything set up in the same environment

File: docs/source/flow_setup.rst (hunk 3 above)
Comment (lucfisc) [reply docs/source/flow_setup.rst]: @AboudyKreidieh Deleted it, will know for the next time ! Thanks for your help

File: docs/source/flow_setup.rst (hunk 2 above)
Comment (eugenevinitsky) [reply docs/source/flow_setup.rst]: Okay, should be good to go after this. Can you add the dockerfile that you used to create the docker image? Would be useful to have in the docker folder as a record if we need to remake it!

**Comment by lucfisc** [comment]
Added the dockerfile, publish the docker to the docker hub. RLLIB is installed, RLLAB is installed , FLOW is installed. Documentation is up to date

**Comment by eugenevinitsky** [comment]
Thanks! Just to check, @lucfisc , ready to merge?

**Pull request merged**